module1/homework/grayscaleImages/main.cpp

```diff
@@ -1,51 +1,119 @@
+#include <algorithm>
 #include <array>
 #include <forward_list>
-
-// TODO: include
-
+#include <functional>
+#include <iostream>
+#include <utility>
+#include "compression.hpp"
```

Comment:
Compression.hpp has array inclusion,so you do not need it here. And I think you should make new line between standard library includes and header include.

Reply:
Done, you have right, I am wondering, why clang-format does not instert this additional line 🤔